app/Models/User.php

```diff
@@ -455,9 +451,9 @@ public function getDynamicContext(): string
 
     public function assignTeam($teamId): void
     {
-        $this->teams()->detach();
+        $this->team()->dissociate()->save();
 
-        $this->teams()->attach($teamId);
+        $this->team()->associate($teamId)->save();
```

Comment:
Because this is not just a `BelongsTo` do we need to do the `dissociate` first here? I could be wrong, but I think the  `associate` may just overwrite, right?